service/src/main/java/uk/nhs/adaptors/gp2gp/common/service/WebClientFilterService.java

```diff
@@ -224,4 +238,43 @@ private static ExchangeFilterFunction logResponse() {
             return Mono.just(response);
         });
     }
+
+    private static Mono<ClientResponse> getInternalServerErrorException(ClientResponse clientResponse, RequestType requestType) {
+        return clientResponse.bodyToMono(String.class).flatMap(body -> {
+            String exceptionMessage;
+
+            if (body.isEmpty()) {
+                exceptionMessage = String.format(REQUEST_EXCEPTION_MESSAGE, requestType, clientResponse.statusCode());
+            } else {
+                exceptionMessage = String.format(REQUEST_EXCEPTION_MESSAGE, requestType, body);
+            }
+
+            return Mono.error(new GpcServerErrorException(exceptionMessage));
```

Comment:
`GpcServerErrorException` is named specific to GPConnect, but this class is "common" this feels strange/unexpected to me

Reply:
The current implementation uses the class as a filter for the MHS and GPC Client. I've refactored this method to handle both MHS and GPC responses.